src/main/java/com/javadeobfuscator/deobfuscator/Deobfuscator.java

```diff
@@ -414,15 +439,15 @@ public boolean runFromConfig(TransformerConfig config) throws Throwable {
     }
 
     public ClassNode assureLoaded(String ref) {
-        ClassNode clazz = classpath.get(ref);
+        ClassNode clazz = classpath.get(ref.replace('/', '.'));
```

Comment:
Classes are stored in this project by their internal name, so this change is wrong.

Reply:
In my test case I got a clear case of `ref` containing `.` which is why I added it. I can revert it if you insist.

Reply:
See
https://github.com/java-deobfuscator/deobfuscator/blob/master/src/main/java/com/javadeobfuscator/deobfuscator/Deobfuscator.java#L109
and
https://github.com/java-deobfuscator/deobfuscator/blob/master/src/main/java/com/javadeobfuscator/deobfuscator/Deobfuscator.java#L132

The map is filled with asm's classname which is the internal full representation.

If you got a call to that method with a dotted class name, it is either a bug of the deobfuscator or some sneaky obfuscation which tries to exploit possible weaknesses of java-deobfuscator.